### ISSUE TITLE
Slow flipping of Save Page Now button: Performance Improvement

### ISSUE THREAD

**Issue by tikhsuP** [internetarchive/wayback-machine-webextension#474]
**Describe the bug**
I am not sure if this is a valid bug or not.
When I hover over the Save Page Now button for the first time, after restarting the extension, it takes longer to flip than the next time. This may be because it takes some time to fetch the data.

**To Reproduce**
Open a page: 
1. Open the extension by clicking on the icon in the Toolbar
2. Hover over 'Save Page Now' for the first time, notice the time it takes to flip
3. Hover over 'Save Page Now' again without restarting it, notice the time it takes to flip
4. The first time takes much longer

**Expected behavior**
It should flip quickly.

**Desktop:**
 - OS: Windows 10
 - Browser: Chrome Version 81.0.4044.122

Can we do anything about it?

**Comment by mitchell-frost** [comment]
Can I work on this?

**Comment by tikhsuP** [comment]
Thank you @mitchell-frost for your interest. Yes, you may work on this issue.
We plan to cache the data to reduce the flipping delay. Please feel free to ask if you need any help.

**Comment by mitchell-frost** [comment]
@tikhsuP I can reproduce the bug but I don't understand the data caching. How am I supposed to achieve that?

**Comment by tikhsuP** [comment]
You can understand the implementation of data caching by looking into the declaration and usage of the variable [`fact_checked_data`](https://github.com/internetarchive/wayback-machine-chrome/blob/master/webextension/scripts/background.js#L19).

When the **Fact Checks** option in **General** settings is checked, the **Fact Check** feature uses the same concept to fetch and store data (if available) automatically when a page is loaded. This happens because the [`chrome.tabs.onUpdated.addListener( )`](https://github.com/internetarchive/wayback-machine-chrome/blob/master/webextension/scripts/background.js#L523
)  is fired when a page is loaded, and it calls the [`factCheckPage(tab.id, tab.url)`](https://github.com/internetarchive/wayback-machine-chrome/blob/master/webextension/scripts/background.js#L533).

More on [chrome.tabs.onUpdated.addListener( )](https://developer.chrome.com/extensions/tabs#event-onUpdated)

**Comment by tikhsuP** [comment]
@mitchell-frost, the idea is to fetch and store the data when a page loads. This will make the data available to the extension before the user hovers over the **Save Page Now** button. And there won't be any need to fetch the data on hover, which causes the delay.

**Comment by mitchell-frost** [comment]
@tikhsuP thank you..I am on it !

**Comment by mitchell-frost** [comment]
I made a few changes to the background.js file. The save now button still takes some time on some URLs but it has decreased the time a lot. I changed the [ chrome.tabs.onUpdated.addListener( )](url) function and added [updateWaybackCountBadge(tab.id, tab.url)](url) and [factCheckPage(tab.id, tab.url)](url) while the tab is loading as well.

**Comment by mitchell-frost** [comment]
The code now looks like this:

```
chrome.tabs.onUpdated.addListener((tabId, info, tab) => {
  if (info.status === 'complete') {
    updateWaybackCountBadge(tab.id, tab.url)
    chrome.storage.local.get(['auto_archive', 'fact_check'], (event) => {
      // fact check
      if (event.fact_check === true) {
        factCheckPage(tab.id, tab.url)
      }
      // auto save page
      if (event.auto_archive === true) {
        auto_save(tab.id, tab.url)
      }
    })
  } else if (info.status === 'loading') {
    updateWaybackCountBadge(tab.id, tab.url)
    chrome.storage.local.get(['auto_archive', 'fact_check'], (event) => {
      // fact check
      if (event.fact_check === true) {
        factCheckPage(tab.id, tab.url)
      }
      // auto save page
      if (event.auto_archive === true) {
        auto_save(tab.id, tab.url)
      }
    })
```

Previously the code was:

```
chrome.tabs.onUpdated.addListener((tabId, info, tab) => {
  if (info.status === 'complete') {
    updateWaybackCountBadge(tab.id, tab.url)
    chrome.storage.local.get(['auto_archive', 'fact_check'], (event) => {
      // fact check
      if (event.fact_check === true) {
        factCheckPage(tab.id, tab.url)
      }
      // auto save page
      if (event.auto_archive === true) {
        auto_save(tab.id, tab.url)
      }
    })
  } else if (info.status === 'loading') {
```

**Comment by anishsarangi** [comment]
The issue has been solved in #692 . 
Thank you !
